src/transformers/models/owlv2/convert_owlv2_to_hf.py

```diff
@@ -0,0 +1,331 @@
+    rename_keys.append(("class_head/logit_shift/kernel", "class_head.logit_shift.weight"))
+    rename_keys.append(("class_head/logit_scale/kernel", "class_head.logit_scale.weight"))
+    rename_keys.append(("class_head/logit_scale/bias", "class_head.logit_scale.bias"))
+    rename_keys.append(("class_head/logit_shift/bias", "class_head.logit_shift.bias"))
+    rename_keys.append(("obj_box_head/Dense_0/kernel", "box_head.dense0.weight"))
+    rename_keys.append(("obj_box_head/Dense_0/bias", "box_head.dense0.bias"))
+    rename_keys.append(("obj_box_head/Dense_1/kernel", "box_head.dense1.weight"))
+    rename_keys.append(("obj_box_head/Dense_1/bias", "box_head.dense1.bias"))
+    rename_keys.append(("obj_box_head/Dense_2/kernel", "box_head.dense2.weight"))
+    rename_keys.append(("obj_box_head/Dense_2/bias", "box_head.dense2.bias"))
+
+    # objectness head (only for v2)
+    if "v2" in model_name:
+        rename_keys.append(("objectness_head/Dense_0/kernel", "objectness_head.dense0.weight"))
+        rename_keys.append(("objectness_head/Dense_0/bias", "objectness_head.dense0.bias"))
+        rename_keys.append(("objectness_head/Dense_1/kernel", "objectness_head.dense1.weight"))
+        rename_keys.append(("objectness_head/Dense_1/bias", "objectness_head.dense1.bias"))
+        rename_keys.append(("objectness_head/Dense_2/kernel", "objectness_head.dense2.weight"))
+        rename_keys.append(("objectness_head/Dense_2/bias", "objectness_head.dense2.bias"))
+
+    # fmt: on
+
+    return rename_keys
+
+
+def rename_and_reshape_key(dct, old, new, config):
+    val = dct.pop(old)
+
+    if ("out_proj" in new or "v_proj" in new or "k_proj" in new or "q_proj" in new) and "vision" in new:
+        val = val.reshape(-1, config.vision_config.hidden_size)
+    if ("out_proj" in new or "v_proj" in new or "k_proj" in new or "q_proj" in new) and "text" in new:
+        val = val.reshape(-1, config.text_config.hidden_size)
+
+    if "patch_embedding" in new:
+        print("Reshaping patch embedding... for", new)
+        val = val.transpose(3, 2, 0, 1)
+    elif new.endswith("weight") and "position_embedding" not in new and "token_embedding" not in new:
+        val = val.T
+
+    if new.endswith("bias"):
+        val = val.reshape(-1)
+
+    dct[new] = torch.from_numpy(np.array(val))
+
+
+@torch.no_grad()
+def convert_owlv2_checkpoint(model_name, pytorch_dump_folder_path, push_to_hub):
+    """
+    Copy/paste/tweak model's weights to our OWL-ViT structure.
+    """
+    config = get_owlv2_config(model_name)
+
+    # Load original state dict based on model name
+    model_name_to_checkpoint_path = {
+        "owlvit-base-patch16": "/Users/nielsrogge/Documents/OWL-ViT/clip_vit_b16_6171dab",
+        "owlv2-base-patch16": "/Users/nielsrogge/Documents/OWL-ViT/owl2-b16-960-st-ngrams_c7e1b9a",
+        "owlv2-base-patch16-ensemble": "/Users/nielsrogge/Documents/OWL-ViT/owl2-b16-960-st-ngrams-curated-ft-lvisbase-ens-cold-weight-05_209b65b",
+    }
```

Comment:
MMmm we should try to be agnostic to your path and use an input path to the checkoints